tests/src/JIT/Directed/VectorABI/VectorMgdMgd.cs

```diff
@@ -24,23 +24,218 @@ public static class VectorMgdMgd
+        }
+    }
+    public static bool CheckValue<T>(T value, T expectedValue)
+    {
+        bool returnVal;
+        if (typeof(T) == typeof(float))
+        {
+            returnVal = Math.Abs(((float)(object)value) - ((float)(object)expectedValue)) <= Single.Epsilon;
+        }
+        if (typeof(T) == typeof(double))
+        {
+            returnVal = Math.Abs(((double)(object)value) - ((double)(object)expectedValue)) <= Double.Epsilon;
+        }
+        else
+        {
+            returnVal = value.Equals(expectedValue);
+        }
+        return returnVal;
+    }
+
     public unsafe class HVATests<T> where T : struct
     {
-        // We can have up to 4 vectors in an HVA, and we'll test structs with up to 5 of them
+        // We can have up to 4 vectors in an HVA, and we'll test structs with up to 6 (3x2) of them
         // (to ensure that even those that are too large are handled consistently).
-        // So we need 5 * the element count in the largest (128-bit) vector with the smallest
+        // So we need 6 * the element count in the largest (128-bit) vector with the smallest
         // element type (byte).
         static T[] values;
         static T[] check;
-        private int ElementCount = (Unsafe.SizeOf<Vector128<T>>() / sizeof(byte)) * 5;
+        private int ElementCount = (Unsafe.SizeOf<Vector128<T>>() / sizeof(byte)) * 6;
         public bool isPassing = true;
+        public bool isReflection = false;
+
+        Type[] reflectionParameterTypes;
+        System.Reflection.MethodInfo reflectionMethodInfo;
+        object[] reflectionInvokeArgs;
+
+        ////////////////////////////////////////
+
+        public struct HVA64_01
+        {
+            public Vector64<T> v0;
+        }
+
+        public struct HVA64_02
+        {
+            public Vector64<T> v0;
+            public Vector64<T> v1;
+        }
+
+        public struct HVA64_03
+        {
+            public Vector64<T> v0;
+            public Vector64<T> v1;
+            public Vector64<T> v2;
+        }
+
+        public struct HVA64_04
+        {
+            public Vector64<T> v0;
+            public Vector64<T> v1;
+            public Vector64<T> v2;
+            public Vector64<T> v3;
+        }
+
+        public struct HVA64_05
+        {
+            public Vector64<T> v0;
+            public Vector64<T> v1;
+            public Vector64<T> v2;
+            public Vector64<T> v3;
+            public Vector64<T> v4;
+        }
+
+        private HVA64_01  hva64_01;
+        private HVA64_02  hva64_02;
+        private HVA64_03  hva64_03;
+        private HVA64_04  hva64_04;
+        private HVA64_05  hva64_05;
+
+        ////////////////////////////////////////
+
+        public struct HVA128_01
+        {
+            public Vector128<T> v0;
+        }
 
         public struct HVA128_02
         {
             public Vector128<T> v0;
             public Vector128<T> v1;
         }
 
+        public struct HVA128_03
```

Comment:
Is there a reason that you only went up to 3 elements for the 128-bit vectors?

Reply:
3*2 was 6 registers.  I thought that 4 registers was the limit.

Can we use 8 registers? If so I can go up top 5  (2*5)

Reply:
The 128-bit vector aggregates, which I guess they actually call SVAs (Scalable Vector Aggregates) are passed in the 128-bit vector registers. Also, reading the document now I'm uncertain as to whether there's even a 4-element restriction, as it's not called out explicitly for those the way it is for the HFAs and HVAs. Maybe @sdmaclea can shed some light here?
Here's the relevant part of the doc:

> 4.3.5.2 Homogeneous Short-Vector Aggregates (HVA)
A Homogeneous Short-Vector Aggregate (HVA) is a Homogeneous Aggregate with a Fundamental Data Type
that is a Short-Vector type and at most four uniquely addressable members.
4.3.6 Scalable-Vector Aggregates (SVAs)
A Scalable Vector Aggregate is an aggregate in which the Fundamental Data Type of each member is a Scalable
Vector Type or a Scalable Predicate Type. The aggregate does not need to be homogeneous. For example, an
aggregate that contains a scalable vector of 8-bit elements, a scalable predicate, and a scalable vector of 16-bit
elements is a Scalable Vector Aggregate.

Note that we don't (presently) support the scalable predicate type.

Reply:
Wait I see the 128 size does fit in a register.

Reply:
I will expand the matrix to include 5 x  Vector128's

Reply:
@CarolEidt We have not really started to think about supporting the Scalable Vector extension.  This is different from HVA.  The scalable here refers to a concept of a register which supports implementation specific longer vectors.  An ARM implementation is allowed to use a multiple of 128-bit register length (up to a maximum 2048-bit?) to support the scalable vector operations. On different implementations it could be 128, 512, 1024....

Reply:
@sdmaclea - thanks for the clarification. I was misled in the documents by the term "Short Vector" which I was taking to mean 64-bit vectors, but now I see/recall that it refers to the non-SVE 128-bit vectors as well.